darts/timeseries.py

```diff
@@ -3607,6 +3612,57 @@ def _get_kwargs(transformation, forecasting_safe):
         if isinstance(transforms, dict):
             transforms = [transforms]
 
+        several_transforms = len(transforms) > 1
+
+        # check if some transformations are applied to the same components
+        overlapping_transforms = False
+        transformed_components = set()
+        for tr in transforms:
+            if not isinstance(tr, dict):
+                raise_log(
+                    ValueError("Every entry in `transforms` must be a dictionary"),
+                    logger,
+                )
+            tr_comps = set(tr["components"] if "components" in tr else self.components)
+            if len(transformed_components.intersection(tr_comps)) > 0:
+                overlapping_transforms = True
+            transformed_components = transformed_components.union(tr_comps)
+
+        partial_transforms = transformed_components != set(self.components)
+
+        if keep_old_names and overlapping_transforms:
+            raise_log(
+                ValueError(
+                    "Cannot keep the original components names as some transforms are overlapping "
+                    "(applied to the same components). Set `keep_old_names` to `False`."
+                ),
+                logger,
+            )
+
+        # actually, this could be allowed to allow transformation "in place"?
+        # keep_non_transformed can be changed to False/ignored if the transforms are not partial
+        if keep_old_names and keep_non_transformed:
+            raise_log(
+                ValueError(
+                    "`keep_old_names = True` and `keep_non_transformed = True` is not supported due to "
+                    " the ambiguity."
+                ),
+                logger,
+            )
+
+        new_hierarchy = None
+        convert_hierarchy = False
+        if self.hierarchy:
+            # the partial_transform covers for scenario keep_non_transformed = True
+            if several_transforms or partial_transforms:
+                logger.warning(
+                    "The hierarchy cannot be retained, either because there is more than one transform or "
+                    "because the transform is not applied to all the components of the series."
+                )
+            else:
+                comp_names_map = dict()
```

Comment:
Let's define this dict for all cases (outside the if/else scope). Then we avoid the "this variable might be undefined" warnings further down